examples/hybrid/dbubble3d_invariant_totalenergy.jl

```diff
@@ -0,0 +1,374 @@
+    @. dw -= fω¹² × fu¹² # Covariant3Vector on faces
+    @. duₕ -= If2c(fω¹² × fu³)
+
+    # Needed for 3D:
+    @. duₕ -=
+        cω³ × Geometry.Contravariant12Vector(Geometry.Covariant123Vector(cuₕ))
+
+    ce = @. cρe / cρ
+    #cp = @. pressure(cρ, ce, cuvw, z) #TODO: the pressure function doesn't work (broadcast error)
+    cI = @. ce - Φ(z) - (norm(cuvw)^2) / 2
+    cT = @. cI / C_v + T_0
+    cp = @. cρ * R_d * cT
+
+    @. duₕ -= hgrad(cp) / cρ
+    vgradc2f = Operators.GradientC2F(
+        bottom = Operators.SetGradient(Geometry.Covariant3Vector(0.0)),
+        top = Operators.SetGradient(Geometry.Covariant3Vector(0.0)),
+    )
+    @. dw -= vgradc2f(cp) / Ic2f(cρ)
+
+    cE = @. (norm(cuvw)^2) / 2 + Φ(z)
+    @. duₕ -= hgrad(cE)
+    @. dw -= vgradc2f(cE)
+
+    # 3) potential temperature
+
+    @. dρe -= hdiv(cuvw * (cρe + cp))
+    @. dρe -= vdivf2c(fw * Ic2f(cρe + cp))
+    @. dρe -= vdivf2c(Ic2f(cuₕ * (cρe + cp)))
+
+    fcc = Operators.FluxCorrectionC2C(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+    fcf = Operators.FluxCorrectionF2F(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+
+    @. dρ += fcc(fw, cρ)
+    @. dρe += fcc(fw, cρe)
+    # dYc.ρuₕ += fcc(w, Yc.ρuₕ)
+
+    Spaces.weighted_dss!(dY.Yc, comms_ctx_center)
+    Spaces.weighted_dss!(dY.uₕ, comms_ctx_center)
+    Spaces.weighted_dss!(dY.w, comms_ctx_face)
+    return dY
+end
+
+dYdt = similar(Y);
+for p in 1:nprocs
+    if pid == p
+        #println("pid = $pid; typeof(Y) = $(typeof(Y))")
+        #println("pid = $pid; typeof(Y.Yc) = $(typeof(Y.Yc))")
+        #println("pid = $pid; size(parent(getfield(Y, :values))) = $(size(parent(getfield(Y, :values))))")
+    end
+    ClimaComms.barrier(comms_ctx_center)
+end
+#rhs_invariant!(dYdt, Y, parameters, 0.0);
+rhs_invariant!(dYdt, Y, nothing, 0.0);
+# run!
+using OrdinaryDiffEq
+Δt = 0.050
+#prob = ODEProblem(rhs_invariant!, Y, (0.0, 700.0), parameters)
+prob = ODEProblem(rhs_invariant!, Y, (0.0, 700.0))
+integrator = OrdinaryDiffEq.init(
+    prob,
+    SSPRK33(),
+    dt = Δt,
+    saveat = 10.0,
+    progress = true,
+    progress_message = (dt, u, p, t) -> t,
+);
+
+if haskey(ENV, "CI_PERF_SKIP_RUN") # for performance analysis
+    throw(:exit_profile)
+end
+
+sol_invariant = @timev OrdinaryDiffEq.solve!(integrator)
+
+sol_global = []
+if usempi
+    for sol_step in sol_invariant.u
+        sol_step_values_global =
+            DataLayouts.gather(comms_ctx, Fields.field_values(sol_step))
+        if ClimaComms.iamroot(Context)
+            sol_step_global = Fields.Field(sol_step_values_global, global_space)
+            push!(sol_global, sol_step_global)
+        end
+    end
+end
+
+
+ENV["GKSwstype"] = "nul"
+import Plots
+Plots.GRBackend()
+
+dir = "bubble3d_invariant_etot"
+path = joinpath(@__DIR__, "output", dir)
+mkpath(path)
+
+#=
+# post-processing
+Es = [sum(Y.Yc.ρe) for u in sol_invariant.u]
```

Comment:
@akshaysridhar : Should this be 
```
Es = [sum(u.Yc.ρe) for u in sol_invariant.u]
````

Reply:
Yes. Likewise in the corresponding 2D example outputs. Thanks @sriharshakandala